htdocs/user/card.php

```diff
@@ -448,19 +448,15 @@
 				$object->lang = GETPOST('default_lang', 'aZ09');
 
 				// Do we update also ->entity ?
-				if (!empty($conf->multicompany->enabled)) {	// If multicompany is not enabled, we never update the entity of a user.
-					if (!empty($conf->global->MULTICOMPANY_TRANSVERSE_MODE)) {
-						$object->entity = 1; // all users are in master entity
+				if (!empty($conf->multicompany->enabled && $user->entity == 0 && !empty($user->admin))) {	// If multicompany is not enabled, we never update the entity of a user.
+					if (GETPOST('superadmin', 'int')) {
```

Comment:
We should not rely on a value from GETPOST to force the entity.
According to your comment "// If multicompany is not enabled, we never update the entity of a user.", if this is true, why no just commenting the line 
$object->entity = 0;
If we must never update the field, we must just not set the value when we are in such a case.